### ISSUE TITLE
Deprecate 'Post author' block

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#53427]
The Post Author block combines the display of the author name, avatar, bio, and 'byline'. It's an older block that is fairly prohibitive in terms of design expression due to all the pieces being locked in position. 

<img width="1444" alt="Screenshot 2023-08-08 at 15 11 10" src="https://github.com/WordPress/gutenberg/assets/846565/e5711d2f-3175-4ecf-97d6-93fc36dc2ba8">

Modern themes tend to make use of the newer Post Author Name, Avatar, and Post Author Biography blocks, combined with layout tools like Row and Stack to gain greater control over the display of this information.

To avoid users falling into the trap of using an outdated block it might be worth deprecating Post Author (IE not surfacing it in the block inserter(s)).

If the prescribed layout is deemed to be of value a core pattern could be included which is constructed from the aforementioned atomic blocks.

**Comment by jameskoster** [comment]
Here's a pattern that could effectively replace Post Autho:

```
<!-- wp:group {"layout":{"type":"flex","flexWrap":"nowrap","justifyContent":"left","verticalAlignment":"top"}} -->
<div class="wp-block-group"><!-- wp:avatar {"size":48} /-->

<!-- wp:group {"style":{"layout":{"selfStretch":"fill","flexSize":null}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"left"}} -->
<div class="wp-block-group"><!-- wp:paragraph {"fontSize":"small"} -->
<p class="has-small-font-size">Write a byline...</p>
<!-- /wp:paragraph -->

<!-- wp:post-author-name {"fontSize":"medium"} /-->

<!-- wp:post-author-biography {"style":{"layout":{"selfStretch":"fit","flexSize":null}},"fontSize":"small"} /--></div>
<!-- /wp:group --></div>
<!-- /wp:group -->
```

**Comment by t-hamano** [comment]
I think this is related to #35596, but I agree with deprecating the Post Author block.

As for Byline, there is no block to display this on its own, but since it is [rendered as just a `p` element](https://github.com/WordPress/gutenberg/blob/b23b6450292c38bc1a0dec8e905203a886f79335/packages/block-library/src/post-author/index.php#L55) in the Post Author block, I believe that a paragraph block can be substituted without going to the trouble of adding a dedicated block.

**Comment by jasmussen** [comment]
No strong opinions, if the block was versatile it could have convenience factor, similar to how Media & Text has. Though if we keep it, is there any reason to call it "Post Author", as opposed to just "Author"? The prefix seems limiting, and presumably it can be used in more than just that one CPT.

**Comment by jameskoster** [comment]
A pattern can be just as convenient, but with the benefit of flexibility. I can't really think of a scenario where it'd be beneficial for a user to use this block.

> is there any reason to call it "Post Author", as opposed to just "Author"?

This applies to all 'template tag' blocks as discussed in https://github.com/WordPress/gutenberg/issues/26856. I'd be in favor of removing the "Post" prefix wholesale. I think it's already been done for Avatar and Title so we're in an awkward middle-ground right now.

**Comment by richtabor** [comment]
> This applies to all 'template tag' blocks as discussed in https://github.com/WordPress/gutenberg/issues/26856. I'd be in favor of removing the "Post" prefix wholesale

Same.

**Comment by t-hamano** [comment]
In #53492, the "Post" prefix was removed from block titles. Can we consider deprecating this block again?

**Comment by jameskoster** [comment]
We really should. That byline feature is confusing. Theme authors should be using the primitive blocks/patterns to make it easier for end users to customise the appearance.

**Comment by t-hamano** [comment]
I would like to work on this issue. If we want to deprecate the Post Author block, just update the `title`, `description`, and `supports.inserter` in `block.json`.

[Here](https://github.com/WordPress/gutenberg/blob/1be856b0d0833828326beef8bb0c16e39d8115f1/packages/block-library/src/comment-author-avatar/block.json#L6-L9) is an example of the Comment Author Avatar block, which has already been deprecated:

I think what we need to consider is whether we want to provide a block pattern equivalent to the Post Author block.

There are [not many patterns in the core](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-patterns.php#L23-L29). Personally, I think the pattern should be provided in the Patterns Directory and this block should just be deprecated in core and Gutenberg.

**Comment by jameskoster** [comment]
> Personally, I think the pattern should be provided in the Patterns Directory and this block should just be deprecated in core and Gutenberg

I agree. There could even be a dedicated category for these types of patterns – there are so many interesting ways to express this content.

**Comment by t-hamano** [comment]
Oops, I noticed that the deprecated Text Column block provides a transformation to Columns block.

- https://github.com/WordPress/gutenberg/pull/9364

We may need to provide a similar transformation for users who have already inserted the Post Author block into their content.

I'd like to build on the markup suggested in [this comment](https://github.com/WordPress/gutenberg/issues/53427#issuecomment-1669716539) and come up with some transformation logic to preserve block style and content as much as possible.
